### ISSUE TITLE
Dimensional consistency on batch is not being checked

### ISSUE THREAD

**Issue by dudanogueira** [weaviate/weaviate#4499]
### How to reproduce this bug?

Run this:

```python
import weaviate
client = weaviate.connect_to_local()
client.collections.delete("MyCollection")
collection = client.collections.create("MyCollection")
content = [
    {"name": "Duda", "vector":[1,2,3,4,5]},
    {"name": "Bob", "vector":[1,2,3,4,5,6,7,8,9,10]},
    {"name": "Byron", "vector":[1,2,3,4,5,6]}
]
with collection.batch.dynamic() as batch:
    for data_row in content:
        batch.add_object(
            properties={"name": data_row["name"]},
            vector=data_row["vector"]
        )
if client.batch.failed_objects:
    print(client.batch.failed_objects)

for i in collection.query.fetch_objects(include_vector=True).objects:
    print(i.properties, i.vector)
```
output:

> {'name': 'Duda'} {'default': [1.0, 2.0, 3.0, 4.0, 5.0]}
>{'name': 'Bob'} {'default': [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]}
>{'name': 'Byron'} {'default': [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]}

### What is the expected behavior?

That code should fail, as we have a dimensions mismatch, for example, this will fail:
```python
collection.data.insert({"name": "Parker"}, vector=[1,2,3,4,5,6,7])
```

> UnexpectedStatusCodeError: Object was not added! Unexpected status code: 500, with response body: {'error': [{'message': 'put object: import into index mycollection: put local object: shard="z20cFOKQ5NNK": Validate vector index for 8fb4d727-1628-4a2d-a31f-44769fb6a108: new node has a vector with length 7. Existing nodes have vectors with length 5'}]}.
PS: the vector used as validation will change for each run.

### What is the actual behavior?

The objects get ingested, and have mismatched dimensional vectors that can lead to unexpected behaviors

### Server Version

1.24.3

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by dudanogueira** [comment]
Also, some more checks needs to be done and raised if it's the case:

Some discussions here:
https://forum.weaviate.io/t/collection-does-not-exist-exception-in-python-v4/2518/6

_thanks ggapac_

**if `AUTOSCHEMA_ENABLED: 'false'` and the collection doesn't exist, it should raise an error.**
like in 

```python
new = client.collections.get("NonExistantCollection")
new.data.insert({"text": "this will fail"})
```

this will not raise any errors

test_col = client.collections.get("NonExistant")

```python
with client.batch.fixed_size(batch_size=100) as batch:
    for i in range(100):
        properties_dict = {
            'col1': "test1",
            'col2': "test2"
        }

        batch.add_object(
            collection="Test",
            properties=properties_dict,
            vector=[0, 1, 2]
        )
```
Also, this will not raise errors (passing an array instead of text)

```python
with collection.batch.dynamic() as batch:
    batch.add_object(
        # this should fail and get in failed_objects
        properties={"text": ["an example"]}
    )
    batch.add_object(
        # it will also allow mismatching dimensions
        properties={"text": "5dimensions"},
        vector=[1,2,3,4,5]
    )
    batch.add_object(
        properties={"text": "6dimensions"},
        vector=[1,2,3,4,5,6]
    )
print(client.batch.failed_objects)
```

**Comment by dudanogueira** [comment]
Note. This is with ASYNC_INDEXING Off
somewhat related: https://github.com/weaviate/weaviate/issues/5082

**Comment by dudanogueira** [comment]
Closing as this is now fixed in 1.26.3
